Application/Telegram/TelegramCommunicationService.cs

```diff
@@ -36,24 +36,51 @@ public async Task ExecuteAsync(Update update)
                 await _telegraBotClient.SendTextMessageAsync(update.Message.Chat.Id, "You are not authorized to use this bot");
                 return;
             }
-                
 
-            //await _telegraBotClient.SendChatActionAsync(update.Message.Chat.Id, ChatAction.Typing);
-            switch (update.Type)
+            try
+            {
+                //await _telegraBotClient.SendChatActionAsync(update.Message.Chat.Id, ChatAction.Typing);
+                switch (update.Type)
+                {
+                    case UpdateType.CallbackQuery:
+                        if (update.CallbackQuery.Data.Contains("BACK"))
+                        {
+                            var state = update.CallbackQuery.Data.Substring(4);
+                            UserState testState = (UserState)Enum.Parse(typeof(UserState), state);
+                            UserStateStorage.UserStateUpdate(update.CallbackQuery.From.Id,
+                                testState);
+                        }
+                        await UpdateCallbackQuery.Handle(update, _telegraBotClient, _mediator);
+                        return;
+                    case UpdateType.Message:
+                        await UpdateMessage.Handle(update, _telegraBotClient, _mediator);
+                        return;
+                }
+            }
+            catch (NotFoundException exeption)
+            {
+                switch (update.Type)
```

Comment:
same switch twice, you can move to separate method

Reply:
done